org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -652,7 +675,11 @@ private PublishDiagnosticJob(ISchedulingRule rule) {
 
 		@Override
 		public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
-			return publishDiagnostics(monitor);
+			long startTime = System.nanoTime();
+			IStatus status = publishDiagnostics(monitor);
```

Comment:
If the latest diagnostic job is cancelled, is it necessary to use its elapsedtime to update the window?

Reply:
Yes, I should check the cancel status.